components/UserContext.js

```diff
@@ -5,8 +5,10 @@ export const UserContext = createContext();
 
 export function UserContextProvider({children}) {
   const [user, setUser] = useLocalStorage("user", "notSet");
+  const [id, setId] = useLocalStorage("id", "notSet");
+
```

Comment:
instead of `"notSet"` you should use `null` to indicate a missing value. 
Is `id` the id of the current user? If so, you should consider having a user object (e.g. {name:"Hans Wurst", id:"abjfhskfjh235dj"}) instead of separate state values.